build.gradle

```diff
@@ -32,6 +34,54 @@ compileJava {
 	options.release = 11
 }
 
+javadoc {
+	/* We don't actually need the full javadoc for anything, so just hijack this task for
+	 * building the client javadocs. If we ever need the full javadocs make a new task for the
+	 * client java docs
+	 */
+	/* TODO DOCS currently java-common and auth links don't work. ROI for fixing them is low.
+	 *           Ideally, in the future make github actions publish docs on release and ref them
+	 *           here.
+	 *           https://github.com/marketplace/actions/deploy-publish-javadoc
+	 *           https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-gradle-registry
+	 */
+	/* TODO DOCS the current sdk documenation looks like invalid html to javadoc
+	 *           need to go through and remove
+	 *           also some spots with < / > that need to be wrapped with {@code } in internal code
+	 */
+	failOnError = false
+	options {
+		links "https://docs.oracle.com/en/java/javase/11/docs/api/"
+		links "https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-core/2.9.9/"
+		links "https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/2.9.9/"
```

Comment:
May I know why we need to include them here?

Reply:
you know what, we don't actually need the jackson links. When the java common docs were built into the workspace docs they were needed since they had references to jackson, but if we do it the right way and host the java docs then these links will be needed in java common, not here